go/vt/schema/online_ddl.go

```diff
@@ -68,42 +76,126 @@ func validateWalk(node sqlparser.SQLNode, allowForeignKeys bool) (kontinue bool,
 	return false, nil
 }
 
-// OnlineDDLStatus is an indicator to a online DDL status
-type OnlineDDLStatus string
-
-const (
-	OnlineDDLStatusRequested OnlineDDLStatus = "requested"
-	OnlineDDLStatusCancelled OnlineDDLStatus = "cancelled"
-	OnlineDDLStatusQueued    OnlineDDLStatus = "queued"
-	OnlineDDLStatusReady     OnlineDDLStatus = "ready"
-	OnlineDDLStatusRunning   OnlineDDLStatus = "running"
-	OnlineDDLStatusComplete  OnlineDDLStatus = "complete"
-	OnlineDDLStatusFailed    OnlineDDLStatus = "failed"
-)
-
 // OnlineDDL encapsulates the relevant information in an online schema change request
 type OnlineDDL struct {
-	Keyspace string      `json:"keyspace,omitempty"`
-	Table    string      `json:"table,omitempty"`
-	Schema   string      `json:"schema,omitempty"`
-	SQL      string      `json:"sql,omitempty"`
-	UUID     string      `json:"uuid,omitempty"`
-	Strategy DDLStrategy `json:"strategy,omitempty"`
-	Options  string      `json:"options,omitempty"`
+	// m is used to provide concurrent access to ReadyToComplete logic. Callers
+	// should use the IsReadyToComplete and MarkReadyToComplete methods to
+	// safely modify the value of that field rather than accessing it through
+	// the embedded protobuf message directly.
+	m sync.Mutex
```

Comment:
I'm wary of adding a mutex member of `OnlineDDL`. Throughout the existing code we pass `OnlineDDL` by reference and by value, and this does not play well with mutex members as their behavior can be inconsistent. `OnlineDDL` was designed to be a data-only object. I'd say we can perhaps enforce users to use `sync/atomic` `Bool` or similar?